### ISSUE TITLE
docs/wasm build failing on M1 Mac

### ISSUE THREAD

**Issue by jhellerstein** [hydro-project/hydroflow#734]
I would be happy just disabling wasm on this platform if that's the fix. In fact my preference would be to limit the core hydroflow repo to stuff that can be build with rustup goodness; anything with bigger toolchains (wasm, node/npm) should go in other repos.

```
(base) website_playground]> CARGO_CFG_HYDROFLOW_GENERATE_DOCS="1" wasm-pack build
[INFO]: 🎯  Checking for the Wasm target...
[INFO]: 🌀  Compiling to Wasm...
   Compiling proc-macro2 v1.0.56
   Compiling quote v1.0.26
   Compiling unicode-ident v1.0.8
   Compiling syn v1.0.109
   Compiling autocfg v1.1.0
   Compiling memchr v2.5.0
   Compiling serde_derive v1.0.160
   Compiling serde v1.0.160
   Compiling libc v0.2.141
   Compiling cfg-if v1.0.0
   Compiling once_cell v1.17.1
   Compiling log v0.4.17
   Compiling regex-syntax v0.6.29
   Compiling indexmap v1.9.3
   Compiling version_check v0.9.4
   Compiling hashbrown v0.12.3
   Compiling serde_json v1.0.96
   Compiling itoa v1.0.6
   Compiling ryu v1.0.13
   Compiling cc v1.0.79
   Compiling either v1.8.1
   Compiling thiserror v1.0.40
   Compiling prettyplease v0.2.4
   Compiling tinyvec_macros v0.1.1
   Compiling tinyvec v1.6.0
   Compiling anyhow v1.0.70
   Compiling dirs-sys v0.3.7
   Compiling aho-corasick v0.7.20
   Compiling slotmap v1.0.6
   Compiling syn v2.0.14
   Compiling proc-macro-error-attr v1.0.4
   Compiling percent-encoding v2.2.0
   Compiling tree-sitter v0.20.10
   Compiling unicode-bidi v0.3.13
   Compiling bitflags v1.3.2
   Compiling unicode-normalization v0.1.22
   Compiling io-lifetimes v1.0.10
   Compiling form_urlencoded v1.1.0
   Compiling regex v1.7.3
   Compiling dirs v3.0.2
   Compiling proc-macro-error v1.0.4
   Compiling idna v0.3.0
   Compiling semver v1.0.17
   Compiling wasm-bindgen-shared v0.2.84
   Compiling rustix v0.37.11
   Compiling unicode-width v0.1.10
   Compiling tree-sitter-loader v0.20.0
   Compiling core-foundation-sys v0.8.4
   Compiling core-foundation v0.9.3
   Compiling url v2.3.1
   Compiling textwrap v0.11.0
   Compiling atty v0.2.14
   Compiling errno v0.3.1
   Compiling libloading v0.7.4
   Compiling ascii v1.1.0
   Compiling chunked_transfer v1.4.1
   Compiling httpdate v1.0.2
   Compiling utf8-width v0.1.6
   Compiling strsim v0.8.0
   Compiling same-file v1.0.6
   Compiling vec_map v0.8.2
   Compiling ansi_term v0.12.1
   Compiling walkdir v2.3.3
   Compiling tiny_http v0.12.0
   Compiling html-escape v0.2.13
   Compiling webbrowser v0.8.9
   Compiling clap v2.34.0
   Compiling which v4.4.0
   Compiling fastrand v1.9.0
   Compiling difference v2.0.0
   Compiling rustc-hash v1.1.0
   Compiling lazy_static v1.4.0
   Compiling glob v0.3.1
   Compiling bumpalo v3.12.0
   Compiling smallbitvec v2.5.1
   Compiling tempfile v3.5.0
   Compiling futures-core v0.3.28
   Compiling rust-sitter-common v0.3.3
   Compiling wasm-bindgen-backend v0.2.84
   Compiling syn-inline-mod v0.5.0
   Compiling pin-project-lite v0.2.9
   Compiling wasm-bindgen v0.2.84
   Compiling futures-sink v0.3.28
   Compiling slab v0.4.8
   Compiling winnow v0.4.1
   Compiling futures-task v0.3.28
   Compiling wasm-bindgen-macro-support v0.2.84
   Compiling toml_datetime v0.6.1
   Compiling futures-channel v0.3.28
   Compiling thiserror-impl v1.0.40
   Compiling hydroflow_lang v0.2.0 (/Users/jmh/code/hydroflow/hydroflow_lang)
   Compiling toml_edit v0.19.8
   Compiling c2rust-bitfields-derive v0.2.1
   Compiling auto_impl v1.0.1
   Compiling wasm-bindgen-macro v0.2.84
   Compiling rust-sitter-macro v0.3.3
   Compiling c2rust-bitfields v0.3.0
   Compiling tree-sitter-tags v0.20.2
   Compiling tree-sitter-highlight v0.20.1
   Compiling tokio v1.27.0
   Compiling futures-util v0.3.28
   Compiling tree-sitter-c2rust v0.20.10
   Compiling js-sys v0.3.61
   Compiling proc-macro-crate v1.3.1
   Compiling futures-macro v0.3.28
   Compiling tokio-macros v2.0.0
   Compiling tracing-core v0.1.30
   Compiling itertools v0.10.5
   Compiling bytes v1.4.0
   Compiling futures-io v0.3.28
   Compiling pin-utils v0.1.0
   Compiling tracing v0.1.37
   Compiling rust-sitter v0.3.3
   Compiling toml v0.5.11
   Compiling tree-sitter-config v0.19.0
   Compiling tree-sitter-cli v0.20.8
   Compiling heck v0.4.1
   Compiling ref-cast v1.0.16
   Compiling sealed v0.5.0
   Compiling futures-executor v0.3.28
   Compiling tokio-util v0.7.7
   Compiling ref-cast-impl v1.0.16
   Compiling cc-traits v2.0.0
   Compiling variadics v0.0.2 (/Users/jmh/code/hydroflow/variadics)
   Compiling pusherator v0.0.1 (/Users/jmh/code/hydroflow/pusherator)
   Compiling rust-sitter-tool v0.3.3
   Compiling lattices v0.2.0 (/Users/jmh/code/hydroflow/lattices)
   Compiling tokio-stream v0.1.12
   Compiling futures v0.3.28
   Compiling hydroflow_macro v0.2.0 (/Users/jmh/code/hydroflow/hydroflow_macro)
   Compiling bincode v1.3.3
   Compiling getrandom v0.2.9
   Compiling byteorder v1.4.3
   Compiling smallvec v1.10.0
   Compiling serde-wasm-bindgen v0.4.5
   Compiling web-sys v0.3.61
   Compiling console_error_panic_hook v0.1.7
   Compiling hydroflow_datalog_core v0.2.0 (/Users/jmh/code/hydroflow/hydroflow_datalog_core)
The following warnings were emitted during compilation:

warning: error: unable to create target: 'No available targets are compatible with triple "wasm32-unknown-unknown"'
warning: 1 error generated.

error: failed to run custom build command for `hydroflow_datalog_core v0.2.0 (/Users/jmh/code/hydroflow/hydroflow_datalog_core)`

Caused by:
  process didn't exit successfully: `/Users/jmh/code/hydroflow/target/release/build/hydroflow_datalog_core-d49e2f54296263fb/build-script-build` (exit status: 1)
  --- stdout
  cargo:rerun-if-changed=src
  TARGET = Some("wasm32-unknown-unknown")
  OPT_LEVEL = Some("3")
  HOST = Some("aarch64-apple-darwin")
  cargo:rerun-if-env-changed=CC_wasm32-unknown-unknown
  CC_wasm32-unknown-unknown = None
  cargo:rerun-if-env-changed=CC_wasm32_unknown_unknown
  CC_wasm32_unknown_unknown = None
  cargo:rerun-if-env-changed=TARGET_CC
  TARGET_CC = None
  cargo:rerun-if-env-changed=CC
  CC = None
  cargo:rerun-if-env-changed=CFLAGS_wasm32-unknown-unknown
  CFLAGS_wasm32-unknown-unknown = None
  cargo:rerun-if-env-changed=CFLAGS_wasm32_unknown_unknown
  CFLAGS_wasm32_unknown_unknown = None
  cargo:rerun-if-env-changed=TARGET_CFLAGS
  TARGET_CFLAGS = None
  cargo:rerun-if-env-changed=CFLAGS
  CFLAGS = None
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("false")
  cargo:rerun-if-env-changed=CC_wasm32-unknown-unknown
  CC_wasm32-unknown-unknown = None
  cargo:rerun-if-env-changed=CC_wasm32_unknown_unknown
  CC_wasm32_unknown_unknown = None
  cargo:rerun-if-env-changed=TARGET_CC
  TARGET_CC = None
  cargo:rerun-if-env-changed=CC
  CC = None
  cargo:rerun-if-env-changed=CFLAGS_wasm32-unknown-unknown
  CFLAGS_wasm32-unknown-unknown = None
  cargo:rerun-if-env-changed=CFLAGS_wasm32_unknown_unknown
  CFLAGS_wasm32_unknown_unknown = None
  cargo:rerun-if-env-changed=TARGET_CFLAGS
  TARGET_CFLAGS = None
  cargo:rerun-if-env-changed=CFLAGS
  CFLAGS = None
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  cargo:rerun-if-env-changed=CC_wasm32-unknown-unknown
  CC_wasm32-unknown-unknown = None
  cargo:rerun-if-env-changed=CC_wasm32_unknown_unknown
  CC_wasm32_unknown_unknown = None
  cargo:rerun-if-env-changed=TARGET_CC
  TARGET_CC = None
  cargo:rerun-if-env-changed=CC
  CC = None
  cargo:rerun-if-env-changed=CFLAGS_wasm32-unknown-unknown
  CFLAGS_wasm32-unknown-unknown = None
  cargo:rerun-if-env-changed=CFLAGS_wasm32_unknown_unknown
  CFLAGS_wasm32_unknown_unknown = None
  cargo:rerun-if-env-changed=TARGET_CFLAGS
  TARGET_CFLAGS = None
  cargo:rerun-if-env-changed=CFLAGS
  CFLAGS = None
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  cargo:rerun-if-env-changed=CC_wasm32-unknown-unknown
  CC_wasm32-unknown-unknown = None
  cargo:rerun-if-env-changed=CC_wasm32_unknown_unknown
  CC_wasm32_unknown_unknown = None
  cargo:rerun-if-env-changed=TARGET_CC
  TARGET_CC = None
  cargo:rerun-if-env-changed=CC
  CC = None
  cargo:rerun-if-env-changed=CFLAGS_wasm32-unknown-unknown
  CFLAGS_wasm32-unknown-unknown = None
  cargo:rerun-if-env-changed=CFLAGS_wasm32_unknown_unknown
  CFLAGS_wasm32_unknown_unknown = None
  cargo:rerun-if-env-changed=TARGET_CFLAGS
  TARGET_CFLAGS = None
  cargo:rerun-if-env-changed=CFLAGS
  CFLAGS = None
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  running: "clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=wasm32-unknown-unknown" "-I" "/var/folders/xf/1chdbv2j30l5249snmfhxx6w0000gs/T/grammarKXsxAB" "-I" "/var/folders/xf/1chdbv2j30l5249snmfhxx6w0000gs/T/grammarKXsxAB/sysroot" "-Wall" "-Wextra" "-o" "/Users/jmh/code/hydroflow/target/wasm32-unknown-unknown/release/build/hydroflow_datalog_core-1e913045f2d1b1a1/out/5b31b87752059a55-parser.o" "-c" "/var/folders/xf/1chdbv2j30l5249snmfhxx6w0000gs/T/grammarKXsxAB/parser.c"
  cargo:warning=error: unable to create target: 'No available targets are compatible with triple "wasm32-unknown-unknown"'
  cargo:warning=1 error generated.
  exit status: 1

  --- stderr


  error occurred: Command "clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=wasm32-unknown-unknown" "-I" "/var/folders/xf/1chdbv2j30l5249snmfhxx6w0000gs/T/grammarKXsxAB" "-I" "/var/folders/xf/1chdbv2j30l5249snmfhxx6w0000gs/T/grammarKXsxAB/sysroot" "-Wall" "-Wextra" "-o" "/Users/jmh/code/hydroflow/target/wasm32-unknown-unknown/release/build/hydroflow_datalog_core-1e913045f2d1b1a1/out/5b31b87752059a55-parser.o" "-c" "/var/folders/xf/1chdbv2j30l5249snmfhxx6w0000gs/T/grammarKXsxAB/parser.c" with args "clang" did not execute successfully (status code exit status: 1).


warning: build failed, waiting for other jobs to finish...
Error: Compiling your crate to WebAssembly failed
Caused by: Compiling your crate to WebAssembly failed
Caused by: failed to execute `cargo build`: exited with exit status: 101
  full command: cd "/Users/jmh/code/hydroflow/website_playground" && "cargo" "build" "--lib" "--release" "--target" "wasm32-unknown-unknown"
(base) website_playground]> 
```

**Comment by shadaj** [comment]
Ah in this case you just need `rustup target add wasm32-unknown-unknown`. I'll add this to the README

**Comment by jhellerstein** [comment]
Needs to be tested and closed

**Comment by jhellerstein** [comment]
resolved by #812 and #807
